scipy/stats/meson.build

```diff
@@ -0,0 +1,101 @@
+  '-Wno-unused-variable']
+
+subdir('tests')
+
+statlib_lib = static_library('statlib_lib',
+    ['statlib/ansari.f',
+     'statlib/spearman.f',
+     'statlib/swilk.f'])
+
+statlib_module = custom_target('statlib_module',
+  output : ['statlib-f2pywrappers.f', 'statlibmodule.c'],
+  input : 'statlib.pyf',
+  command : [py3, generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@'],
+)
+
+statlib = py3.extension_module('statlib',
+  [statlib_module],
+  c_args: numpy_nodepr_api,
+  fortran_args: fortran_ignore_warnings,
+  link_with : statlib_lib,
+  dependencies : [py3_dep],
+  include_directories : [inc_np, inc_f2py],
+  install : true,
+  subdir : 'scipy/stats')
+
+# TODO: Add c_args: numpy_nodepr_api 
+_stats = py3.extension_module('_stats',
+  ['_stats.pyx'],
+  dependencies : [py3_dep],
+  include_directories : [inc_np],
+  install : true,
+  subdir : 'scipy/stats')
+
+mvn_module = custom_target('mvn_module',
+  output : ['mvn-f2pywrappers.f', 'mvnmodule.c'],
+  input : 'mvn.pyf',
+  command : [py3, generate_f2pymod, '@INPUT@', '-o', '@OUTDIR@'],
+)
+
+mvn = py3.extension_module('mvn',
+  [mvn_module, 'mvndst.f'],
+  c_args: numpy_nodepr_api,
+  fortran_args: fortran_ignore_warnings,
+  dependencies : [py3_dep],
+  include_directories : [inc_np, inc_f2py],
+  install : true,
+  subdir : 'scipy/stats')
+
+_sobol = py3.extension_module('_sobol',
+  ['_sobol.pyx'],
+  c_args: numpy_nodepr_api,
+  dependencies : [py3_dep],
+  include_directories : [inc_np],
+  install : true,
+  subdir : 'scipy/stats')
+
+py3.install_sources(
+  ['_sobol_direction_numbers.npz'],
+  pure : false,
+  subdir : 'scipy/stats'
+)
+
+pthread = dependency('threads')
+run_command(cython, '--cplus', '_qmc_cy.pyx')
+_qmc_cy = py3.extension_module('_qmc_cy',
+  ['_qmc_cy.cpp'],
+  dependencies : [py3_dep, pthread],
+  include_directories : [inc_np],
+  install : true,
+  subdir : 'scipy/stats')
+
+py3.install_sources(
+  ['biasedurn.pxd'],
+  pure : false,
+  subdir : 'scipy/stats'
+)
+
+npymath_path = incdir_numpy / '..' / 'lib'
+npymath_lib = meson.get_compiler('c').find_library('npymath', dirs: npymath_path)
+npyrandom_path = incdir_numpy / '..' / '..' / 'random' / 'lib'
+npyrandom_lib = meson.get_compiler('c').find_library('npyrandom', dirs: npyrandom_path)
+incdir_biasedurn = incdir_numpy / '..' / '..' / 'random' / 'lib'
+inc_baisedurn = include_directories(incdir_biasedurn)
+run_command(py3, '_generate_pyx.py')
+run_command(cython, '--cplus', 'biasedurn.pyx', '-o', 'biasedurn.cpp')
+m_lib = meson.get_compiler('c').find_library('m')
+run_command(py3, 
+  ['-c', 'import os; os.environ["R_BUILD"] = None;'])
+biasedurn = py3.extension_module('biasedurn',
+  ['biasedurn/stocR.h', 
+   'biasedurn.cpp', 'biasedurn/impls.cpp',
+   'biasedurn/fnchyppr.cpp', 'biasedurn/wnchyppr.cpp',
+   'biasedurn/stoc1.cpp', 'biasedurn/stoc3.cpp'],
+   c_args: numpy_nodepr_api,
+    include_directories: [inc_np, inc_baisedurn],
+    dependencies: [py3_dep, npyrandom_lib, npymath_lib, m_lib]
+)
```

Comment:
This is giving two errors, 
```bash
../scipy/scipy/stats/biasedurn/impls.cpp: In member function 'double StochasticLib1::Normal(double, double)':
../scipy/scipy/stats/biasedurn/impls.cpp:8:10: error: 'next_normal' was not declared in this scope
    8 |   return next_normal(m, s);
      |          ^~~~~~~~~~~
```

and,

```bash
../scipy/scipy/stats/biasedurn.cpp: In function 'int __pyx_pf_5scipy_5stats_9biasedurn_17_PyStochasticLib3___cinit__(__pyx_obj_5scipy_5stats_9biasedurn__PyStochasticLib3*)':
../scipy/scipy/stats/biasedurn.cpp:3590:30: error: 'class StochasticLib3' has no member named 'next_double'
 3590 |   __pyx_v_self->c_sl3.get()->next_double = (&__pyx_f_5scipy_5stats_9biasedurn_next_double);
      |                              ^~~~~~~~~~~
../scipy/scipy/stats/biasedurn.cpp:3599:30: error: 'class StochasticLib3' has no member named 'next_normal'
 3599 |   __pyx_v_self->c_sl3.get()->next_normal = (&__pyx_f_5scipy_5stats_9biasedurn_next_normal);
```

Reply:
Any ideas how to fix the above?

Reply:
`R_BUILD` flag was needed and things worked. I Googled about this flag but it seems like it is something specific to scipy. @rgommers What does this flag do?

Reply:
Interesting, I've never seen it before. The `biasedurn` extension is new, so I think it came in when that was added. I'll have a look at it.

Reply:
`biasedurn` is vendored code that can be used standalone, or as an R extension. And now we're building a Python extension, but the C++ code wasn't written with that in mind. It seems like the Python extension has the same requirements as the R extension (which makes sense), and different from standalone C++. So `R_BUILD` is the toggle between those.

It doesn't appear anywhere else in the SciPy code, so after reading the above you can forget it again:)

Reply:
Thanks for the clarification. I will confirm if `R_BUILD` is a C++ flag or not. Though putting it in `cpp_args` gets things going but still I will double check.